cmake/thirdParties.cmake

```
@@ -55,17 +28,46 @@ if(HDF5_FOUND)
     elseif(TARGET hdf5)
         # target from hdf5 1.8 config
         target_link_libraries(HDF5::HDF5 INTERFACE hdf5)
-        if(NOT HDF5_USE_STATIC_LIBRARIES)
-            target_compile_definitions(HDF5::HDF5 INTERFACE "H5_BUILT_AS_DYNAMIC_LIB")
-        endif()
     else()
         # results from CMake FindHDF5
-        target_include_directories(HDF5::HDF5 INTERFACE "${HDF5_INCLUDE_DIRS}")
-        target_link_libraries(HDF5::HDF5 INTERFACE "${HDF5_LIBRARIES}")
-        target_compile_definitions(HDF5::HDF5 INTERFACE "${HDF5_DEFINITIONS}")
+        set_target_properties(HDF5::HDF5 PROPERTIES
+            INTERFACE_INCLUDE_DIRECTORIES "${HDF5_INCLUDE_DIRS}"
+            INTERFACE_LINK_LIBRARIES "${HDF5_LIBRARIES}"
+        )
+    endif()
+    if(NOT HDF5_USE_STATIC_LIBRARIES)
+        set_target_properties(HDF5::HDF5 PROPERTIES
+            INTERFACE_COMPILE_DEFINITIONS "H5_BUILT_AS_DYNAMIC_LIB"
+        )
     endif()
 endif()
 
 if(NOT HAVE_HDF5 AND MATIO_MAT73)
     message(FATAL_ERROR "MAT73 requires HDF5")
 endif()
+
+
+macro(matio_create_zlib target)
+    add_library(ZLIB::ZLIB INTERFACE IMPORTED)
+    target_link_libraries(ZLIB::ZLIB INTERFACE ${target})
+    set(ZLIB_FOUND TRUE)
+endmacro()
+
+if(MATIO_WITH_ZLIB)
+    if(HDF5_USE_STATIC_LIBRARIES AND TARGET zlib-static)
+        matio_create_zlib(zlib-static)
+    elseif(HDF5_USE_STATIC_LIBRARIES AND TARGET hdf5::zlib-static)
+        matio_create_zlib(hdf5::zlib-static)
+    elseif(NOT HDF5_USE_STATIC_LIBRARIES AND TARGET zlib-shared)
+        matio_create_zlib(zlib-shared)
+    elseif(NOT HDF5_USE_STATIC_LIBRARIES AND TARGET hdf5::zlib-shared)
+        matio_create_zlib(hdf5::zlib-shared)
+    elseif(TARGET zlib)
+        matio_create_zlib(zlib)
+    else()
+        find_package(ZLIB 1.2.3)
```

```suggestion
        find_package(ZLIB 1.2.3)
    endif()
```

endif missingor not?